### PR TITLE
compute: reconcile task states with external queues

### LINKED ISSUE #28: Watch for jobs that never start
Possibly closely related to #25

### LINKED ISSUE #118: Report when jobs cannot be scheduled on HTCondor, Slurm, etc.
It's tricky to know when a job isn't starting on HTCondor because it doesn't match any workers. You really have to know a lot about administering both a Funnel cluster and an HTCondor cluster. You need to find the Funnel logs and know about `condor_q -analyze`

There's probably something nice we can do to report scheduling information from HTCondor

### PR DESCRIPTION
**Pull request by adamstruck**
based off of #439 

https://github.com/adamstruck/funnel/compare/compute-cancel...adamstruck:compute-reconcile

closes #28 
closes #118 (minus grid engine)

### CODE REVIEW

File: compute/htcondor/backend.go [hunk 1]
```diff
@@ -32,3 +44,96 @@ func extractID(in string) string {
...
+				output = append(output, &compute.HPCTaskState{ID: id, TESState: tes.SystemError, State: state, Reason: "Funnel worker exited with non-zero status"})
+			}
+		}
+	}
+	return output, nil
+}
+
+func checkIdleStatus(id string) error {
+	cmd := exec.Command("condor_q", "-analyze", id)
+	stdout, _ := cmd.Output()
```
Comment (buchanae) [inline compute/htcondor/backend.go]: Might as well return the error. Who knows, we might need it later.

File: compute/pbs/backend.go [hunk 2]
```diff
@@ -27,3 +38,78 @@ func NewBackend(conf config.Config, reader tes.ReadOnlyServer, writer events.Wri
...
+
+	idSet := make(map[string]interface{})
+	for _, i := range ids {
+		idSet[i] = nil
+	}
+
+	cmd := exec.Command("qstat", "-x")
+	stdout, err := cmd.Output()
+	if err != nil {
+		return nil, err
```
Comment (buchanae) [inline compute/pbs/backend.go]: can you wrap these errors? I've noticed with other pieces of code with unwrapped errors that I struggle to hunt down the source line in funnel

**Comment by adamstruck** [comment]
Since this PR increases the likelihood of concurrent state updates I recommend merging #482 first.

File: tests/slurm.config.yml [hunk 3]
```diff
@@ -1 +1,4 @@
 Compute: "slurm"
+Slurm:
+  DisableReconciler: false
```
Comment (buchanae) [inline tests/slurm.config.yml]: Sorry to be picky, but these should match the actual default config, i.e. they should be true.

**Pull request merged**